src/VisualStudio/Core/Def/Implementation/LanguageClient/AbstractLiveShareLanguageServerClient.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.Threading;
+using Nerdbank.Streams;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService
+{
+    internal abstract class AbstractLiveShareLanguageServerClient : ILanguageClient
```

Comment:
Why "LiveShare" in the name? Because it's used for other stuff maybe too?

Reply:
Right - can rename now that it's being used by razor as well.